src/middlewared/middlewared/plugins/zfs.py

```diff
@@ -547,6 +547,14 @@ def path_to_dataset(self, path):
 
         return ds_name
 
+    def child_dataset_names(self, path):
+        # return child datasets given a dataset `path`.
+        try:
+            with libzfs.ZFS() as zfs:
+                return [child.name for child in zfs.get_dataset_by_path(path).children]
+        except libzfs.ZFSException:
+            return []
```

Comment:
Why are we not raising the error and giving the illusion that there are no child datasets ?

Reply:
Good point, I'll remove the seatbelt.